sdk/appconfiguration/app-configuration/src/appConfigurationClient.ts

```diff
@@ -508,28 +531,40 @@ export function getGeneratedClientOptions(
       ...retryPolicies,
       ...defaults,
     ],
-    userAgentHeaderName: getUserAgentHeaderName(internalAppConfigOptions)
```

Comment:
The reason we are not using the user agent policy from core-http is because we need a different header name?

Reply:
Yes, and we're not sure if the header name should be set like that everywhere.